include/deal.II/lac/computation.h

```diff
@@ -0,0 +1,639 @@
+  {
+    return *this = *this + Computation<Range>(offset);
+  }
+
+  /**
+   * Subract a constant @p offset (of the @p Range space) from the result
+   * of a computation.
+   */
+  Computation<Range> &operator-=(const Range &offset)
+  {
+    return *this = *this - Computation<Range>(offset);
+  }
+
+  /**
+   * Scalar multiplication of the Computation with a @p number.
+   */
+  Computation<Range> &operator*=(typename Range::value_type number)
+  {
+    return *this = *this * number;
+  }
+  //@}
+};
+
+
+/**
+ * @name Vector space operations
+ */
+//@{
+
+/**
+ * \relates Computation
+ *
+ * Addition of two Computation objects @p first_comp and @p second_comp given by
+ * vector space addition of the corresponding results.
+ *
+ * @ingroup LAOperators
+ */
+template <typename Range>
+Computation<Range>
+operator+(const Computation<Range> &first_comp,
+          const Computation<Range> &second_comp)
+{
+  Computation<Range> return_comp;
+
+  return_comp.reinit_vector = first_comp.reinit_vector;
+
+  // ensure to have valid computation objects by catching first_comp and
+  // second_comp by value
+
+  return_comp.result = [first_comp, second_comp](Range &v)
+  {
+    first_comp.result(v);
+    second_comp.result_add(v);
+  };
+
+  return_comp.result_add = [first_comp, second_comp](Range &v)
+  {
+    first_comp.result_add(v);
+    second_comp.result_add(v);
+  };
+
+  return return_comp;
+}
+
+/**
+ * \relates Computation
+ *
+ * Subtraction of two Computation objects @p first_comp and @p second_comp
+ * given by vector space addition of the corresponding results.
+ *
+ * @ingroup LAOperators
+ */
+template <typename Range>
+Computation<Range>
+operator-(const Computation<Range> &first_comp,
+          const Computation<Range> &second_comp)
+{
+  Computation<Range> return_comp;
+
+  return_comp.reinit_vector = first_comp.reinit_vector;
+
+  // ensure to have valid computation objects by catching first_comp and
+  // second_comp by value
+
+  return_comp.result = [first_comp, second_comp](Range &v)
+  {
+    first_comp.result(v);
+    v *= -1.;
+    second_comp.result_add(v);
+    v *= -1.;
+  };
+
+  return_comp.result_add = [first_comp, second_comp](Range &v)
+  {
+    first_comp.result_add(v);
+    v *= -1.;
+    second_comp.result_add(v);
+    v *= -1.;
+  };
+
+  return return_comp;
+}
+
+/**
+ * \relates Computation
+ *
+ * Scalar multiplication of a Computation objects @p comp with a scalar @p number
+ * given by a scaling Computation result with @p number.
+ *
+ * @ingroup LAOperators
+ */
+template <typename Range>
+Computation<Range>
+operator*(const Computation<Range> &comp,
+          typename Range::value_type number)
+{
+  Computation<Range> return_comp;
+
+  return_comp.reinit_vector = comp.reinit_vector;
+
+  return_comp.result = [comp, number](Range &v)
+  {
+    comp.result(v);
+    v *= number;
+  };
+
+  return_comp.result_add = [comp, number](Range &v)
+  {
+    comp.result_add(v);
+    v *= number;
```

Comment:
this is wrong since it also multiplies whatever was in `v` before by the given number.